app/models/miq_expression.rb

```diff
@@ -681,16 +681,25 @@ def _to_sql(exp, tz)
     when "not like"
       field = Field.parse(exp[operator]["field"])
       clause = field.does_not_match("%#{exp[operator]["value"]}%").to_sql
-    when "and", "or"
-      operands = exp[operator].collect do|operand|
-        o = _to_sql(operand, tz) if operand.present?
-        o.blank? ? nil : o
-      end.compact
-      if operands.length > 1
-        clause = "(" + operands.join(" #{self.class.normalize_sql_operator(operator)} ") + ")"
-      elsif operands.length == 1 # Operands may have been stripped out during pre-processing
-        clause = "(" + operands.first + ")"
+    when "and"
+      operands = exp[operator].each_with_object([]) do |operand, result|
+        next if operand.blank?
+        sql = _to_sql(operand, tz)
+        next if sql.blank?
+        result << sql
+      end
+      clause = Arel::Nodes::And.new(operands.collect { |o| Arel::Nodes::SqlLiteral.new(o) }).to_sql
+    when "or"
```

Comment:
@imtayadeway Other than the Arel class `Arel::Nodes::And` vs. `Arel::Nodes::Or`, I would expected the code building them to be the same. I can see that this all works from the tests but it's not obvious why they need to be handled differently. Can you give a brief explanation? Thanks!